src/utils/messages.ts

```diff
@@ -0,0 +1,31 @@
+export enum Queue {
+  WALLET,
+  GAMEENGINE,
+}
+
+export class Message {
+  queue: Queue;
+  body: string;
+
+  constructor({queue, body}: {queue: Queue, body: string}) {
+    this.queue = queue;
+    this.body = body;
+  }
+
+  get isWallet(): boolean {
+    return this.queue === Queue.WALLET;
+  }
+}
+
+export function decodeMessage(message: string) : Message {
+  const [queueName, body] = message.split('-');
```

Comment:
It would probably be easier to add the `queueName` as a property in firebase. That said, I don't think that should be a blocker for this PR.